### PR TITLE
[PT] Strip pruned model

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Add strip function for pruned model
Simplify strip to avoid build graph for `IN_PLACE` and `DQ` format.

### Related tickets

176026

### Tests

https://github.com/openvinotoolkit/nncf/actions/runs/19017695102

### CODE REVIEW

**Comment by ljaljushkin** [comment]
Nice feature! 
Will be extremely useful to reduce graph building overhead for MoE models with a lot of linear layers.

File: tests/torch2/function_hook/pruning/rb/test_strip.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        # Set mask
+        pruning_module.mask[0] *= -1
+        pruned_weight = pruning_module(pruned_model.conv.weight)
+
+    striped_model = nncf.strip(pruned_model, strip_format=nncf.StripFormat.IN_PLACE, do_copy=False)
+    hook_storage = get_hook_storage(striped_model)
+
+    assert not list(hook_storage.named_hooks())
+    assert torch.equal(striped_model.conv.weight, pruned_weight)
+    assert torch.count_nonzero(striped_model.conv.weight) == 54
```
Comment (ljaljushkin) [inline tests/torch2/function_hook/pruning/rb/test_strip.py:43]: I'd also check count for binary mask or after applying it

File: src/nncf/torch/function_hook/pruning/strip.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if hook_type != "post_hooks" or port_id != 0:
+            msg = f"Unexpected place of SparsityBinaryMask: {hook_type=}, {op_name=}, {port_id=}"
+            raise nncf.InternalError(msg)
+
+        module_name, weight_attr_name = split_const_name(op_name)
+        module = get_module_by_name(module_name, model)
+        weight_param = getattr(module, weight_attr_name)
+
+        weight_param.requires_grad = False
+        weight_param.data = hook_module(weight_param)
```
Comment (ljaljushkin) [inline src/nncf/torch/function_hook/pruning/strip.py:53]: maybe `with torch.no_grad():` is also needed

Comment (AlexanderDokuchaev) [reply src/nncf/torch/function_hook/pruning/strip.py:53]: https://github.com/openvinotoolkit/nncf/pull/3716/files#diff-596206cbb3e55fa75a2947ee6daf7d3dbddb5e545e0f990278ecbfbfa43bcee6R27
Used decorator `@torch.no_grad()`

File: tests/torch2/function_hook/pruning/rb/test_strip.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply tests/torch2/function_hook/pruning/rb/test_strip.py:43]: No think
Mask set directly by `pruning_module.mask[0] *= -1` 
`RBPruningMask` and `binary_mask` are tested in test_modules.py

File: src/nncf/torch/function_hook/pruning/strip.py (hunk 2 above)
Comment (ljaljushkin) [reply src/nncf/torch/function_hook/pruning/strip.py:53]: Oh, I didn't notice for pruning it's global for the function, because for quantization it's local via context manager. different strip method look really the same, maybe some common function would avoid different implementations

File: tests/torch2/function_hook/pruning/rb/test_strip.py (hunk 1 above)
Comment (ljaljushkin) [reply tests/torch2/function_hook/pruning/rb/test_strip.py:43]: yes, agree. but I'm more concerned about initial weights. what if initialization is changed. the test doesn't check initial state.

**Review by ljaljushkin — APPROVED** [review]
